### DISCUSSION TITLE
Is this a `libicu` bug, or am I using it incorrectly?

### DISCUSSION THREAD

**Discussion by ghost** [msys2/MINGW-packages#20147] [Q&A]
```
#include <stdio.h>
#include <stdlib.h>
#include <unicode/ustring.h>

void print_hex_u8(const char *const str) {
	for (size_t i = 0; i < strlen(str); ++i) {
		printf("%02x ", (unsigned char) str[i]);
	}
	printf("\n");
}

void print_hex_u16(const char16_t *const str, size_t str_len) {
	for (size_t i = 0; i < str_len; ++i) {
		printf("%02x ", str[i]);
	}
	printf("\n");
}

int main(void) {
	char[] u8str = "Здравствуйте";
	int u16str_len;
	UErrorCode error;
	u_strFromUTF8(NULL, 0, &u16str_len, u8str, -1, &error); // calculate u16str_len
	UChar* u16str = malloc(sizeof(UChar) * (u16str_len + 1));
	u_strFromUTF8(u16str, u16str_len + 1, &u16str_len, u8str, -1, &error); // actual conversion
	print_hex_u8(u8str);
	print_hex_u16(u16str, u16str_len);
	return 0;
}
```

**Comment by Biswa96** [comment]
Please provide the expected and actual output.

**Reply by ghost** [reply]
Compile the above code and see it yourself. Too hard for me to describe, since the only thing printed on the screen is the result of `print_hex_u8(u8str);`, this means `u16str` is empty.

Let me elaborate more: `U_SUCCESS(error)` is false (this is debug code so it's not included in the code above), this means calculating `u16str_len` failed. Yes, it failed even before the actual conversion.

The error code is 2, which is `U_MISSING_RESOURCE_ERROR`. What that resource is, I don't know. Google search is useless. Most of the results point to ICU Data, which is a shared library and I should have it already as I used `icu-config --ldflags`.

If you have Linux, please compile the above code to see if it works or not. This will tell if my code is wrong or your `libicu` is buggy.

**Reply by ghost** [reply]
If I ditch `icu-config` and link with the `libicu*.dll` files on `/clang64/bin` directly, then `print_hex_u16` will print gibberish, and the result will change each time the program is run. If I remember it right, `U_SUCCESS(error)` is true. This means according to `libicu`, the conversion is success. In no way I could consider such result as a success conversion. So, I'm really out of ideas.
